torchci/scripts/queue_alert.py

```diff
@@ -0,0 +1,192 @@
+import argparse
+import json
+import os
+import re
+from pathlib import Path
+from typing import Any, Dict, List, NamedTuple
+
+import requests
+import rockset  # type: ignore[import]
+from check_alerts import (
+    GRAPHQL_URL,
+    REPO_OWNER,
+    TEST_INFRA_REPO_NAME,
+    clear_alerts,
+    create_issue,
+    headers,
+    update_issue,
+)
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+PROD_VERSIONS_FILE = REPO_ROOT / "torchci" / "rockset" / "prodVersions.json"
+
+QUEUE_ALERT_LABEL = "queue-alert"
+
+MAX_HOURS = 2.5
```

Comment:
The threshold might be too tight here, as I see a 3.4 hours queue for Windows today:

![Screenshot 2023-05-23 at 18 03 47](https://github.com/pytorch/test-infra/assets/475357/8d2fd954-99b6-42ef-a9d1-f7891e612914)